fable-test-app/src/pages/Home.tsx

```diff
@@ -1,16 +1,74 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import axios from 'axios';
 
 // Components (internal)
-import { DateModified, Heading, Text } from '../components';
+import { DateModified, Heading, NextHoliday, Text, Button } from '../components';
+import { holidayObject } from '../utils/constants';
 
 const Home: React.FC = () => {
+  const currentDate = new Date().getTime();
+  const [nextFederal, setNextFederal] = useState<holidayObject>();
+  const [nextNationwide, setNextNationwide] = useState<holidayObject>();
+
+  useEffect(() => {
+    axios.get('https://canada-holidays.ca/api/v1/holidays')
+    .then(({ data }) => {
+
+      // Assign next federal holiday
+      let fedAssigned = false;
+      data.holidays.map((holiday: holidayObject) => {
+        if (holiday.federal === 1) {
+          const holidayDate = new Date (holiday.date).getTime();
+          if (holidayDate > currentDate && !fedAssigned) {
```

Comment:
Small suggestion to do the boolean check first so it doesn't have to do the second check if it's already been assigned
```suggestion
          if (!fedAssigned && holidayDate > currentDate) {
```